builtin/credential/cert/path_login.go

```diff
@@ -175,6 +175,15 @@ func (b *backend) verifyCredentials(req *logical.Request, d *framework.FieldData
 			if tCert.SerialNumber.Cmp(clientCert.SerialNumber) == 0 &&
 				bytes.Equal(tCert.AuthorityKeyId, clientCert.AuthorityKeyId) &&
 				b.matchesConstraints(clientCert, trustedNonCA.Certificates, trustedNonCA) {
+
+				// We are not looking for trusted chains here since this is a
+				// non-CA cert. But validating the connection state detects
+				// expired certificates.
+				_, err := validateConnState(roots, connState)
```

Comment:
I think we can just move this check that is happening after this block above the block and avoid the code duplication.